src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -466,42 +439,145 @@ void string_constraint_generatort::add_axioms_for_correct_number_format(
     /// index < length && correct => is_digit(str[index])
     implies_exprt character_at_index_is_digit(
       binary_relation_exprt(index_expr, ID_lt, str.length()),
-      is_digit_with_radix(str[index], radix_char_type));
+      is_digit_with_radix(str[index], radix_as_char));
     axioms.push_back(character_at_index_is_digit);
   }
 }
 
 /// Add axioms connecting the characters in the input string to the value of the
-/// output integer
-/// \param x: symbol for abs(integer represented by str)
+/// output integer. It is constructive because it gives a formula for x in terms
+/// of the characters in str.
+/// \param x: the integer represented by str
 /// \param type: the type for x
 /// \param char_type: the type to use for characters
 /// \param index_type: the type to use for indexes
 /// \param str: input string
 /// \param max_string_length: the maximum length str can have
 /// \param radix: the radix, with the same type as x
-/// \return an expression for the integer represented by the string
-exprt string_constraint_generatort::add_axioms_for_characters_in_integer_string(
-  const symbol_exprt& x,
+/// \param radix_ul: the radix as an unsigned long, or 0 if that can't be
+///   determined
+void string_constraint_generatort::add_axioms_for_characters_in_integer_string_constructive(
```

Comment:
The constructive approach won, which means the function name now fits on one line. It also has the side effect that we don't use `power_exprt` any more.